src/main/java/org/jabref/gui/externalfiles/ImportHandler.java

```diff
@@ -195,9 +198,10 @@ public void importEntries(List<BibEntry> entries) {
 
     public void importCleanedEntries(List<BibEntry> entries) {
         bibDatabaseContext.getDatabase().insertEntries(entries);
-        generateKeys(entries);
         setAutomaticFields(entries);
         addToGroups(entries, stateManager.getSelectedGroups(bibDatabaseContext));
+        finalizedEntries.addAll(entries);
+        generateKeys(finalizedEntries);
```

Comment:
Why the list? What happens on a second import of other entries? Will the citation keys of the previous entries be replaced?